app/src/components/DependencyCards.tsx

```diff
@@ -0,0 +1,86 @@
+import {
+  ActionIcon,
+  Center,
+  createStyles,
+  em,
+  Grid,
+  Paper,
+  Text,
+  Tooltip,
+  getBreakpointValue,
+  rem
+} from '@mantine/core';
+import Link from 'next/link';
+import React from 'react';
+import { SquareArrowRight } from 'tabler-icons-react';
+
+interface DependencyInformation {
+  type?: string;
+  link?: string;
+}
+
+const useStyles = createStyles(theme => ({
+  card: {
+    borderRadius: 6,
+    border: `${rem(2)} solid ${theme.colors.gray[3]}`,
+    width: '800px',
+    [`@media (max-width: ${em(getBreakpointValue(theme.breakpoints.lg) - 1)})`]: {
+      width: '100%'
+    }
+  }
+}));
+
+/**
+ * Component which displays all data requirement dependencies for a specified resource
+ * and displays them as resource cards that contain all required information and link to their
+ * respective resources if a link exists
+ */
+function Dependencies(props: { relatedArtifact: fhir4.RelatedArtifact }) {
+  const { classes } = useStyles();
+  const display = props.relatedArtifact.display;
+  const resourceLink = props.relatedArtifact.resource;
+
+  let dependencyInfo: DependencyInformation = {};
+
+  if (resourceLink?.includes('Library')) {
+    const resourceArr: string[] = resourceLink.substring(resourceLink.indexOf('Library')).split('|');
+    dependencyInfo = { type: 'Library', link: resourceArr[0] };
+  } else if (resourceLink?.includes('Measure')) {
+    const resourceArr: string[] = resourceLink.substring(resourceLink.indexOf('Measure')).split('|');
+    dependencyInfo = { type: 'Measure', link: resourceArr[0] };
+  }
```

Comment:
If I understand what this strategy is doing, we are taking the text in brackets:

`http://somethingsomething.com/Library/[SOME TEXT]|0.0.0`

And using that text as the "`id`" that we route to to view the details of a resource, since the way our app is set up, the routing responds to `/[resourceType]/[id]`. Unfortunately, I don't think this strategy will work all of the time. 

Canonicals for FHIR resources can theoretically be anything. They _should_ be of the format that this code works on, but they might not always. I could have a measure whose URL is something like `http://example.com/Measure/MattRules123`, but the actual FHIR Measure resource that exists in the measure repository service has a completely different server-defined `.id` property that is not `MattRules123`.

A way to get around this might be out of scope for your task, so I might suggest we create another task to handle this, but I think we need to add logic to this app that can handle trying to view resource details using the canonical URL of the resource as the identifying information. So a URL like `localhost:3001/Measure?url=http://example.com/Measure/MattRules123` should be able to look up the measure in the service by canonical and then route to it.

OR we just have this be part of the existing search page instead or something. Either way, your code doesn't have to change approaches for now, but I would maybe suggest adding a comment above this block that talks about what I just mentioned and we can have a TODO to address it in later tasking

Reply:
Definitely a great point, I didn't even consider this when first starting on the task! I agree it might be a bit out of scope however, so I added the TODO like you suggested :)